From 74cbfbe81356e195608c54fb6b0302957873401e Mon Sep 17 00:00:00 2001
From: Nick <nick@khadas.com>
Date: Mon, 25 Mar 2019 11:08:51 +0800
Subject: [PATCH 091/101] GX: update TEXT_OFFSET for Amlogic MESON SoC

this allow Amlogic vender u-boot v2015.01 to bootup mainline linux well
---
 arch/arm64/Makefile      | 4 ++++
 arch/arm64/kernel/head.S | 4 ++--
 2 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index ab4ccda07966..92d3e9fbd03a 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -102,8 +102,12 @@ TEXT_OFFSET := $(shell awk "BEGIN {srand(); printf \"0x%06x\n\", \
 		 int(2 * 1024 * 1024 / (2 ^ $(CONFIG_ARM64_PAGE_SHIFT)) * \
 		 rand()) * (2 ^ $(CONFIG_ARM64_PAGE_SHIFT))}")
 else
+ifeq ($(CONFIG_ARCH_MESON),y)
+TEXT_OFFSET := 0x01080000
+else
 TEXT_OFFSET := 0x00080000
 endif
+endif
 
 ifeq ($(CONFIG_KASAN_SW_TAGS), y)
 KASAN_SHADOW_SCALE_SHIFT := 4
diff --git a/arch/arm64/kernel/head.S b/arch/arm64/kernel/head.S
index 989b1944cb71..af9945cfb4db 100644
--- a/arch/arm64/kernel/head.S
+++ b/arch/arm64/kernel/head.S
@@ -40,8 +40,8 @@
 #error TEXT_OFFSET must be at least 4KB aligned
 #elif (PAGE_OFFSET & 0x1fffff) != 0
 #error PAGE_OFFSET must be at least 2MB aligned
-#elif TEXT_OFFSET > 0x1fffff
-#error TEXT_OFFSET must be less than 2MB
+//#elif TEXT_OFFSET > 0x1fffff
+//#error TEXT_OFFSET must be less than 2MB
 #endif
 
 /*
-- 
2.17.1

